### PR TITLE
httpc enable options for async request

### PR DESCRIPTION
**Pull request by Whaileee**

### CODE REVIEW

**Review by u3s — CHANGES_REQUESTED** [review]
please:
- fix failed checks
- add label "team:PS"
- re-request review when above are solved

File: lib/inets/src/http_client/httpc.erl [hunk 1]
```diff
@@ -1263,35 +1264,19 @@ mk_chunkify_fun(ProcessBody) ->
...
-    Body = maybe_format_body(BinBody, Options),
-    case proplists:get_value(full_result, Options, true) of
-	true ->
-	    {ok, {StatusLine, Headers, Body}};
-	false ->
-	    {_, Status, _} = StatusLine,
-	    {ok, {Status, Body}}
+	    {error, Reason};
+	{http, {RequestId, {ok, _} = Result}} ->
+        %%%%%%%%%%%%MOVE TO HTTPC HANDLER 1329
```
Comment (u3s) [inline lib/inets/src/http_client/httpc.erl:1276]: ?

File: lib/inets/test/httpc_SUITE.erl [hunk 2]
```diff
@@ -569,7 +569,7 @@ async(Config) when is_list(Config) ->
 	    {http, Msg} ->
 		ct:fail(Msg)
 	end,
-    inets_test_lib:check_body(binary_to_list(Body)),
+    inets_test_lib:check_body(Body),
```
Comment (u3s) [inline lib/inets/test/httpc_SUITE.erl:572]: should we have some new testcase or updates in some of the existing?

File: lib/inets/src/http_client/httpc.erl (hunk 1 above)
Comment (Whaileee) [reply lib/inets/src/http_client/httpc.erl:1276]: whoops, missed that one

File: lib/inets/test/httpc_SUITE.erl (hunk 2 above)
Comment (Whaileee) [reply lib/inets/test/httpc_SUITE.erl:572]: probably testcase for checking full_result option would be viable

**Comment by Whaileee** [comment]
@github-actions disable-cache

**Comment by Whaileee** [comment]
@github-actions disable-cache

File: lib/inets/src/http_client/httpc_handler.erl [hunk 3]
```diff
@@ -1717,6 +1718,43 @@ format_address({[$[|T], Port}) ->
 format_address(HostPort) ->
     HostPort.
 
+format_answer(Res0, Options) ->
+    FullResult = proplists:get_value(full_result, Options, true),
+    Sync = proplists:get_value(sync, Options, true),
+    do_format_answer(Res0, FullResult, Sync).
+do_format_answer({Ref, StatusLine}, _, Sync) when is_atom(StatusLine) ->
```
Comment (u3s) [inline lib/inets/src/http_client/httpc_handler.erl:1725]: make a newline between function definitions: `format_answer/2` and `do_format_answer/3`

**Review by u3s — COMMENTED** [review]
I have no more comments.

regarding targeting of this change, by default bug fixes should be provided for all supported versions. maybe something to be discussed later.

**Review by IngelaAndin — APPROVED** [review]

**Comment by IngelaAndin** [comment]
We should wait a little to decide where to target this. And keep  internal support  ERIERL-1091 in mind.

**Review by u3s — APPROVED** [review]

**Pull request closed without merging**